Client/mods/deathmatch/logic/luadefs/CLuaVectorGraphicDefs.cpp

```diff
@@ -0,0 +1,318 @@
+                CLuaMain*  pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+
+                if (!pLuaMain || !pVectorGraphic)
+                    return false;
+
+                return pVectorGraphic->SetDocument(pXMLNode);
+            },
+            [funcRef](const bool didLoad) {
+                CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(funcRef.GetLuaVM());
+                if (pLuaMain)
+                {
+                    CLuaArguments arguments;
+                    arguments.PushBoolean(didLoad);
+                    arguments.Call(pLuaMain, funcRef);
+                }
+            });
+
+        return true;
+    }
+
+    return pVectorGraphic->SetDocument(pXMLNode);
+}
+
+CLuaMultiReturn<int, int> CLuaVectorGraphicDefs::SVGGetSize(CClientVectorGraphic* pVectorGraphic)
+{
+    return {(int)pVectorGraphic->GetRenderItem()->m_uiSizeX, (int)pVectorGraphic->GetRenderItem()->m_uiSizeY};
+}
+
+bool CLuaVectorGraphicDefs::SVGSetSize(CClientVectorGraphic* pVectorGraphic, int width, int height, std::optional<CLuaFunctionRef> luaFunctionRef)
+{
+    if (width <= 0 || height <= 0)
+        throw std::invalid_argument("A vector graphic must be atleast 1x1 in size.");
+
+    if (width > 4096 || height > 4096)
+        throw std::invalid_argument("A vector graphic cannot exceed 4096x4096 in size.");
+
+    CVectorGraphicItem* pVectorGraphicItem = pVectorGraphic->GetRenderItem();
+
+    if (luaFunctionRef.has_value() && VERIFY_FUNCTION(luaFunctionRef.value()))
+    {
+        CLuaFunctionRef funcRef = luaFunctionRef.value();
+
+        CLuaShared::GetAsyncTaskScheduler()->PushTask<bool>(
+            [funcRef, pVectorGraphic, pVectorGraphicItem, width, height] {
+                lua_State* luaVM = funcRef.GetLuaVM();
+                CLuaMain*  pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+
+                if (!pLuaMain || !pVectorGraphic)
+                    return false;
+
+                pVectorGraphicItem->Resize(*new CVector2D(width, height));
+
+                if ((int)pVectorGraphicItem->m_uiSizeX != width || (int)pVectorGraphicItem->m_uiSizeY != height)
+                    return false;            // failed to resize
+
+                pVectorGraphic->GetDisplay()->Update();
+                return true;
+            },
+            [funcRef](const bool didLoad) {
+                CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(funcRef.GetLuaVM());
+                if (pLuaMain)
+                {
+                    CLuaArguments arguments;
+                    arguments.PushBoolean(didLoad);
+                    arguments.Call(pLuaMain, funcRef);
+                }
+            });
+
+        return true;
+    }
+
+    pVectorGraphicItem->Resize(*new CVector2D(width, height));
+
+    if ((int)pVectorGraphicItem->m_uiSizeX != width || (int)pVectorGraphicItem->m_uiSizeY != height)
+        return false;            // failed to resize
+
+    pVectorGraphic->GetDisplay()->Update();
+
+    return true;
```

Comment:
The code is currently duplicated twice, once for the callback and once for the non-callback. I would like this to be refactored into a e.g. single private function call.